### PR TITLE
Add flush_immediately to writer.

### LINKED ISSUE #60: `Writer`: Add option to flush buffer more frequently
As the data we are recording with `Writer` is important (e.g. experiment data) it might be useful to have the option to flush the write buffer more frequently. You can change the default buffering behaviour in Python when you call `open()`: https://docs.python.org/3/library/functions.html#open

It may be sensible to flush data with every line written. This can be achieved if the user explicitly calls `flush()` after every write, but currently this means they have to use the private `_file` member. It is probably cleaner all round if they can just set this up when they create the `Writer` object in the first place. We might even want to turn this option on by default; the performance penalty isn't likely to be high.

In a similar vein, I have also been using `os.fsync()` before closing CSV files, which tells the OS to commit the changes to disk immediately. We could do this automatically when the user enables the option to flush with every line.

What do you think @dalonsoa? @cc-a Do you have any thoughts about this?

### PR DESCRIPTION
**Pull request by jamesturner246**
Users can specify that the writer should flush after each write operation by setting the optional `flush_immediately` flag true in the writer constructor. Getters and setters also added.

I'm not sure how to easily test that flush has done it's job with pytest, or even if it's possible.

Closes #60

### CODE REVIEW

File: csvy/writers.py [hunk 1]
```diff
@@ -95,10 +101,22 @@ def close(self) -> None:
...
+        if self._flush_immediately:
+            self._file.flush()
+
+    @property
+    def flush_immediately(self) -> bool:
+        return self._flush_immediately
+
+    @flush_immediately.setter
+    def flush_immediately(self, flag: bool) -> None:
+        self._flush_immediately = flag
```
Comment (dalonsoa) [inline csvy/writers.py:119]: I would not include any setter/getter for this. I wouldn't expect users to change the flush flag midway through the process. If we do want to allow that, just remove the underscore from `self._flush_immediately` and let them change it directly.

**Review by alexdewar — CHANGES_REQUESTED** [review]
What I had in mind for this was to set the `buffering` flag in `open` to 1, which automatically flushes the stream with every line: https://docs.python.org/3/library/functions.html#open

**Comment by jamesturner246** [comment]
Maybe we can stay flexible and accept the integer flag and pass it as-is to open?

**Review by dalonsoa — CHANGES_REQUESTED** [review]
Looks good. I've just suggested to remove the setter/getter part, as I feel it is not necessary. 

About testing, it will be necessary to use mocks and check if `flush` is called.

**Comment by dalonsoa** [comment]
I've just seen @alexdewar comment. Yes, if that's possible, it will be simpler, to be honest.

**Review by dalonsoa — APPROVED** [review]
Shorter and simpler 👍

**Review by alexdewar — APPROVED** [review]
LGTM!

**Comment by dalonsoa** [comment]
@all-contributors please add @jamesturner246 for code

**Pull request merged**